prereise/gather/demanddata/transportation_electrification/smart_charging.py

```diff
@@ -0,0 +1,365 @@
+
+        i = 0
+
+        while i < nd_len:
+            # zero individual block, this for each vehicle per day
+            individual = 0
+
+            # trip amount for each vehicle
+            total_trips = newdata.iloc[
+                i, newdata.columns.get_loc("total vehicle trips")
+            ]
+
+            if data_day[i] == input_day[day_iter]:
+
+                # only home based trips
+                if (
+                    newdata.iloc[i, newdata.columns.get_loc("why from")]
+                    * newdata.iloc[i + total_trips, newdata.columns.get_loc("why to")]
+                    == 1
+                ):
+
+                    # copy one vehicle information to the block
+                    individual = newdata[i : i + total_trips - 1]
+
+                    # 'trip' is used to determine if the loop should end. 'trip' is to count the trip number
+                    # e.g. 1.5-3.3 has 3 segment 1 2 3
+                    trip_num = 0
+
+                    # get the consumption and charging constraints for each trip(hour segment)
+                    charging_consumption = (
+                        []
+                    )  # zero the consumption, this is a vector, the consumption for each trip
+                    rates = (
+                        []
+                    )  # zero the rates of all segments, which will be used as cost function later
+                    elimit = (
+                        []
+                    )  # zero the charging energy limits of all segments, which will be used as the upper bound
+                    seg = []
+
+                    # total_trips * 2 matrix, recording dwelling start and end time.
+                    dwelling_times = []
+
+                    while trip_num < total_trips:
+                        # DC consumption
+                        charging_consumption.append(
+                            individual.iloc[
+                                trip_num, individual.columns.get_loc("Miles traveled")
+                            ]
+                            * kwhmi
+                            * -1
+                        )
+
+                        # dwelling part, energylimit is the charging energy constraint
+                        # get segment, t1, tn, s1, sn, energylimit, extracost from dwell function
+                        dwelling_start_time = individual.iloc[
+                            trip_num,
+                            individual.columns.get_loc("End time (hour decimal)"),
+                        ]
+                        dwelling_length = individual.iloc[
+                            trip_num,
+                            individual.columns.get_loc("Dwell time (hour decimal)"),
+                        ]
+                        total_dwell_period = dwelling_start_time + dwelling_length
+
+                        segment = dwelling.get_segment(
+                            dwelling_start_time, total_dwell_period
+                        )
```

Comment:
It seems that the `dwelling.get_segment` function just calculates `total_dwell_period - dwelling_start_time + 1`. If                         `total_dwell_period = dwelling_start_time + dwelling_length`, then are we essentially just calculating `dwelling_length + 1` here?